test/test_optim.py

```diff
@@ -597,8 +597,16 @@ def _compare_between(self, inputs, models, optimizers, assert_eq_kwargs=None, as
             for input, model, optimizer in zip(inputs, models, optimizers):
                 optimizer.zero_grad()
 
+                if i == 3:
+                    # Freeze a layer to test if the step of this layer in 'fused' or 'foreach'
+                    # is same as the step in 'forloop'.
+                    model[2].requires_grad_(False)
+                if i == 5:
+                    # Unfreeze the layer after 2 iters.
+                    model[2].requires_grad_(True)
+
                 # Test that step behaves as expected (a no-op) when grads are set to None
-                if i != 3:
+                if i != 2:
```

Comment:
was this sufficient to be failing before the change?

Reply:
no. i rollback to the original asgd code (ill step & wrong implementation), but it passes all the tests except {"lambd": 1.0}

Reply:
Setting lambd > 0.1 is sufficient to indicate failure before fixing the bugs.
Should I set the test case to `"lambd": 0.1` or just keep `"lambd": 1`?

Reply:
0.1 seems better lol.

it's ok if it passes all tests other than the lambd config.

Reply:
I think the problem is caused by `torch.as_tensor` (as I described above). I'd like to dig into it, but I only found the type definition in pytorch/torch/_C/_VariableFunctions.pyi. Could you tell me how to find the .h and .cpp files for `torch.as_tensor`?

Reply:
It's defined here pytorch/torch/csrc/utils/tensor_new.cpp but this is not a bug. The PR fix is sufficient with the bigger lambd.